libdrgn/dwarf_index.c

```diff
@@ -583,6 +584,13 @@ static struct drgn_error *read_cu(struct drgn_dwarf_index_cu_buffer *buffer)
 					 &buffer->cu->address_size)))
 		return err;
 
+	// Skip rest of TU to properly place buffer pos
```

Comment:
I don't like splitting this TU specific logic both here and in `index_cu_header_offset`, but I think this is the cleanest way to do it.